packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -60,21 +60,47 @@ const NODE_MANIFEST_FILE_LIMIT = getNodeManifestFileLimit()
  */
 async function findPageOwnedByNode({
   nodeId,
+  fullNode,
   slug,
 }: {
   nodeId: string
+  fullNode: IGatsbyNode
   slug?: string
 }): Promise<{
   page: INodeManifestPage
   foundPageBy: FoundPageBy
 }> {
   const state = store.getState()
   const { pages, nodes } = state
-  const { byNode } = state.queries
-
-  // the default page path is the first page found in
-  // node id to page query tracking
-  let pagePath = byNode?.get(nodeId)?.values()?.next()?.value
+  const { byNode, byConnection, trackedComponents } = state.queries
+
+  const { staticQueryComponents } = state
+  const nodeType = fullNode?.internal?.type
+
+  let pagePath =
+    // the default page path is the first page found in
+    // node id to page query tracking
+    byNode?.get(nodeId)?.values()?.next()?.value ||
```

Comment:
instead of getting the first value — I wonder if sorting the values could yield better results e.g. sort by length so `/` is better than `/a-long-url`

Reply:
I think it would be the same, in either case it's random right? Probably better not to sort as it's less work, especially for large sites where there could be tons of node manifest files.

Reply:
yeah — and this is backup anyways to a site setting the node / page connection directly so 🤷

Reply:
Exactly :) this is a fallback in case we can't find anything more specific, so they atleast see that content somewhere on their site.